src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceRepository.kt

```diff
@@ -1,13 +1,16 @@
 package uk.gov.justice.digital.hmpps.hmppshdcapi.licences
 
+import org.springframework.data.domain.Limit
 import org.springframework.data.domain.Page
 import org.springframework.data.domain.Pageable
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor
 import org.springframework.stereotype.Repository
+import java.time.LocalDateTime
 
 @Repository
 interface LicenceRepository : JpaRepository<Licence, Long>, JpaSpecificationExecutor<Licence> {
   fun findByPrisonNumber(prisonNumber: String, pageable: Pageable): Page<Licence>
   fun findAllByPrisonNumber(prisonNumber: String): List<Licence>
+  fun findAllByDeletedAtAndIdGreaterThanLastProcessedAndOrderByIdAsc(deletedAt: LocalDateTime?, lastProcessed: Long, limit: Limit): Page<Licence>
```

Comment:
Not 100% sure if this is the correct way to do this, following the use of [keywords supported for JPA](https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html) 👀

Reply:
Hoping it does all the following: `select * from licence where id > ${lastIdProcessed} and deleted is null order by id asc limit ${batchSize}` 😅

Reply:
when you write the integration test you should be able to see the generated sql in the logs of the test run

Reply:
(if it doesn't you could write a custom query)